### DISCUSSION TITLE
Crowd getEditableAgent purpose

### DISCUSSION THREAD

**Discussion by caxapexac** [recastnavigation/recastnavigation#544] [Q&A]
Documentation says I mustn't change any of position, velocity etc. of the agent but I have this method to change anything inside, is it safe enough (stable) to e.g. teleport my agent by just changing dtCrowdAgent::npos?

**Comment by caxapexac** [comment]
The question is mainly on the purpose of this function, not about changing position, so reference https://github.com/recastnavigation/recastnavigation/issues/425

**Comment by memononen** [comment]
You should not directly change the agent as you describe. In order to teleport, the best way is to remove and add again the agent.

**Comment by caxapexac** [comment]
> You should not directly change the agent as you describe. In order to teleport, the best way is to remove and add again the agent.

But I've asked about the theoretical usecase of this particular function because documentation lacks of the description. I suppose Ive already said that I know about conflicts on using this for teleport.

I mean, if this function is completely useless (in case we have overload with const return value) maybe there is something to do with removing this function?
